laborator/content/rolul-registrelor-adresare/demo/declaration/declaration.asm

```diff
@@ -0,0 +1,52 @@
+;
+; Author: Gabriel Mocanu <gabi.mocanu98@gmail.com>
+;
+
+%include "../../utils/printf32.asm"
+
+%define ARRAY_SIZE 13
+%define DECIMAL_PLACES 5
+
+; http://www.tortall.net/projects/yasm/manual/html/nasm-pseudop.html
+; db = define byte - 1 byte
+; dw = define word - 2 bytes
+; dd = define double word - 4 bytes
+; dq = define quad word - 8 bytes
+; resb
+; resw
+; resd
+
+section .bss
+    W          resd 1
+    var2       DB ?     ; Declară un octet neinițializat etichetat cu "var2".
+    X          DW ?     ; Declară un cuvânt(2 octeți) neinițializat, etichetat cu "X".
+
+section .data
+
+    num_array dw 76, 12, 65, 19, 781, 671, 431, 761, 782, 12, 91, 25, 9
+    decimal_point db ".",0
+    var        db 64    ; Declară un octet conținând valoarea 64. Etichetează
+                        ; locația de memorie cu "var".
+               db 10    ; Declară un octet neetichetat, inițializat cu 10. Acest
+                        ; octet va fi plasat la adresa (var + 1).
+    Y          dd 3000  ; Declară un cuvânt dublu (4 octeți) cu eticheta "Y",
+                        ; inițializat cu valoarea 3000.
+    Z          dd 1,2,3 ; Declară 3 cuvinte duble (a câte 4 octeți fiecare)
+                        ; incepând cu adresa "Z" și inițializate cu 1, 2, respectiv 3.
+                        ; De exemplu, 3 va fi plasat la adresa (Z + 8).
+    var5           equ 2
+    my_array   times 3 db 8, 9, 10
+
+section .text
+
+extern printf
+global main
+main:
```

Comment:
Nu inteleg continutul main-ului. Ar trebui intr-adevar afisate niste valori ale zonelor de memorie declarate mai sus. (Forma de acum nu imi spune nimic). Imi scapa ceva aici?

Reply:
E doar un demo pentru declarare si afisare a unor valori. Va fi ceva dinamic si interactiv deci main-ul cel mai probabil va fi schimbat in timpul laboratorului cu mai multe exemple din cauza asta nu am luat in calcul sa afisez mai multe lucruri in main

Reply:
Okay, it make sense.